src/agent.spec.ts

```diff
@@ -0,0 +1,374 @@
+        arguments: [1, "metadata", [1, 2]],
+      })
+      .addTraces({
+        function: iface.getFunction("updateAgent"),
+        to: AGENT_REGISTRY_ADDRESS,
+        from: NETHERMIND_DEPLOYER_ADDRESS,
+        arguments: [1, "metadata", [1, 2]],
+      });
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toEqual([
+      Finding.fromObject({
+        name: "UPDATE Agent Detected",
+        description: `UPDATE function called by Nethermind's deployer address`,
+        alertId: "NETHERMIND-2",
+        severity: FindingSeverity.Low,
+        type: FindingType.Info,
+        metadata: {
+          data: "0x",
+          from: NETHERMIND_DEPLOYER_ADDRESS.toLocaleLowerCase(),
+          to: AGENT_REGISTRY_ADDRESS.toLocaleLowerCase(),
+          txHash: "0x",
+        },
+      }),
+      Finding.fromObject({
+        name: "UPDATE Agent Detected",
+        description: `UPDATE function called by Nethermind's deployer address`,
+        alertId: "NETHERMIND-2",
+        severity: FindingSeverity.Low,
+        type: FindingType.Info,
+        metadata: {
+          data: "0x",
+          from: NETHERMIND_DEPLOYER_ADDRESS.toLocaleLowerCase(),
+          to: AGENT_REGISTRY_ADDRESS.toLocaleLowerCase(),
+          txHash: "0x",
+        },
+      }),
+    ]);
+  });
+
+  it("should detect mutliple Disable Agent function calls in traces", async () => {
+    const txEvent = new TestTransactionEvent()
+      .setFrom(DEPLOYER_ADDRESS)
+      .setTo(AGENT_REGISTRY_ADDRESS)
+      .addTraces({
+        function: iface.getFunction("disableAgent"),
+        to: AGENT_REGISTRY_ADDRESS,
+        from: NETHERMIND_DEPLOYER_ADDRESS,
+        arguments: [1, 1],
+      })
+      .addTraces({
+        function: iface.getFunction("disableAgent"),
+        to: AGENT_REGISTRY_ADDRESS,
+        from: NETHERMIND_DEPLOYER_ADDRESS,
+        arguments: [1, 1],
+      });
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toEqual([
+      Finding.fromObject({
+        name: "DISABLE Agent Detected",
+        description: `DISABLE function called by Nethermind's deployer address`,
+        alertId: "NETHERMIND-3",
+        severity: FindingSeverity.Low,
+        type: FindingType.Info,
+        metadata: {
+          data: "0x",
+          from: NETHERMIND_DEPLOYER_ADDRESS.toLocaleLowerCase(),
+          to: AGENT_REGISTRY_ADDRESS.toLocaleLowerCase(),
+          txHash: "0x",
+        },
+      }),
+      Finding.fromObject({
+        name: "DISABLE Agent Detected",
+        description: `DISABLE function called by Nethermind's deployer address`,
+        alertId: "NETHERMIND-3",
+        severity: FindingSeverity.Low,
+        type: FindingType.Info,
+        metadata: {
+          data: "0x",
+          from: NETHERMIND_DEPLOYER_ADDRESS.toLocaleLowerCase(),
+          to: AGENT_REGISTRY_ADDRESS.toLocaleLowerCase(),
+          txHash: "0x",
+        },
+      }),
+    ]);
+  });
```

Comment:
Great test coverage! Would recommend adding test cases of checking for multiple function calls of different functions. And additionally, one that is the same with multiple function calls of different types mix with non-valid function calls, to confirm it only detects the ones that it should and filter out the non-valid ones (e.g. `fooAgent()`).